source/extensions/filters/http/cache/cache_filter_utils.cc

```diff
@@ -0,0 +1,33 @@
+#include "extensions/filters/http/cache/cache_filter_utils.h"
+
+#include "common/common/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+bool CacheFilterUtils::isCacheableRequest(const Http::RequestHeaderMap& headers) {
+  const Http::HeaderEntry* method = headers.Method();
+  const Http::HeaderEntry* forwarded_proto = headers.ForwardedProto();
+  const Http::HeaderValues& header_values = Http::Headers::get();
+  // TODO(toddmgreer): Also serve HEAD requests from cache.
+  // TODO(toddmgreer): Check all the other cache-related headers.
+  return method && forwarded_proto && !headers.Authorization() && headers.Path() &&
```

Comment:
IIUC this line is the substance of the PR. Is that right?

Reply:
Yes it is, the rest is refactoring of this function so that it is testable.